Allow using alternate auth for swift

giovannipizzi opened the pull request:
In our environment, we have a different identity provider (via tokens) to access swift. For this reason, even providing the correct passwords, projects, tenants, ... I wasn't able to connect to my swift installation.

However, the swift library allows for an alternate authentication: https://godoc.org/github.com/ncw/swift#Connection.ApplyEnvironment
where only two variables are needed:
```
OS_STORAGE_URL - storage URL from alternate authentication
OS_AUTH_TOKEN - Auth Token from alternate authentication
```
however currently rclone is always checking the username, password, ... and always trying to authenticate.

In this PR, I check if the two variables above are set from the environment (when `env_auth` is `true`) and in this case skip the authentication and just let the underlying library do its job, by using the alternate authorization.
With this patch, I am now able to connect to my swift installation, after using an external authentication script to get the token.
In the environment, I just need to have something like
```
export OS_AUTH_TOKEN=<MY_TOKEN>
export OS_STORAGE_URL=https://<server>:<port>/v1/AUTH_<project_id>
```

ncw commented:
Great idea!

Can you `go fmt` your code please? That is why the build failed.

Also can you start your commit message with `swift: ` so I know which subsystem this is - thanks.

giovannipizzi commented:
Ok, I did `go fmt` now (sorry, it's my very first time with go :-) - it was already not obvious how to compile this ;-) ).
I did also a few changes: I'm now using `c.Authenticated()` instead of checking that the AuthToken string is not empty; and I also made it possible now to read both variables from the configuration (there is a new variable for auth_token now). I'm not sure where/how to update the docs though... Will you do it?

giovannipizzi commented:
(Found the docs file... I added a section and also updated some outdated output.
I hope changes here are then automatically reflected in the other files, if any (html, ...)

ncw commented:
That looks great, thank you :-)
